test/integration/tests/modules.cpp

```diff
@@ -166,6 +166,10 @@ TEST_F(ZNCTest, AutoAttachModule) {
     ircd.Write(":foo PRIVMSG #znc :hello");
     ircd.ReadUntil("TEST");
     client.ReadUntil("hello");
+    client.Write("PRIVMSG *autoattach :Add * test *");
+    client.ReadUntil("Added to list");
+    client.Write("PRIVMSG *autoattach :Swap 1 2")
+    client.ReadUntil("Rules Swapped");
```

Comment:
I can show you much easier implementation which will still pass this test:

```c++
AddCommand("Swap", "", "",
                   [=](const CString& sLine) { PutModule("Rules Swapped"); });
```

Can you code here the exact issue which you were facing without this new functionality in place? That's the point of having a test.

Reply:
Okay, I am working on expanding this, but the thing I'm not sure how to test is the client correctly not attaching when the ignore rule is in place.